src/app/mobile-app/middleware.ts

```diff
@@ -17,5 +17,13 @@ export const checkMobileAppDetails: RequestHandler = async (req, res) => {
     res.status(apiResponse.status);
   }
 
+  // If we don't have a clientOAuthSessionId in our session then we're dealing with a cross browser callback and core back will give us a clientOAuthSessionId to use instead.
```

Comment:
I'm not very happy about having to have this special case handling in core-front.
It might be better to find a way of having the app callback be handled as a route in core-back so that this can be hidden from core-front. However that would mean changes to journey handling to cope with the cross-browser case having no IpvSession.

Reply:
@blakeyp do you have any thoughts about this? Is it worth the time to remove this?
I think it would require changes to core-back to:
- Accept a parameter to a journey request (for the state)
- Cope with the lack of IPV session - at least for this journey
It would also have the advantage of making this special case handling visible in the journey map rather than hidden 'magic'

Reply:
I think it's definitely worth exploring, could you raise a ticket or spike? It might be something we'd do post go-live as an improvement